includes/class-analytics.php

```diff
@@ -0,0 +1,245 @@
+<?php
+/**
+ * Newspack Analytics features.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Manages Analytics integrations.
+ */
+class Analytics {
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		add_filter( 'googlesitekit_amp_gtag_opt', [ __CLASS__, 'inject_amp_events' ] );
+		add_action( 'wp_footer', [ __CLASS__, 'inject_non_amp_events' ] );
+	}
+
+	/**
+	 * Get data about all events.
+	 *
+	 * An event is largely based on the AMP analytics event spec and should contain the following fields:
+	 *    'id'             => string Event ID (e.g. "popupFormDisplayed").
+	 *    'on'             => string Type of action (e.g. "click" or "form-submit-success").
+	 *    'amp_on'         => string Type of action for AMP if different (e.g. "amp-form-submit-success").
+	 *    'element'        => string CSS selector for element (e.g. '#popup1').
+	 *    'amp_element'    => string CSS selector for AMP version of element if different.
+	 *    'event_name'     => string Name for event in GA (e.g. 'Clicked').
+	 *    'event_label'    => string Label for event in GA (e.g. 'Popup 1')
+	 *    'event_category' => string Category for event in GA (e.g. 'Newspack Announcements').
+	 * There can also be other fields specific for certain events (e.g. 'scrollSpec' for 'scroll' event listener).
+	 */
+	public static function get_events() {
+		$events = [
+			[
+				'id'             => 'socialShareClickedFacebook',
+				'on'             => 'click',
+				'element'        => 'a.share-facebook',
+				'amp_element'    => 'amp-social-share[type="facebook"]',
+				'event_name'     => 'social share',
+				'event_label'    => 'facebook',
+				'event_category' => 'NTG social',
+			],
+			[
+				'id'             => 'socialShareClickedTwitter',
+				'on'             => 'click',
+				'element'        => 'a.share-twitter',
+				'amp_element'    => 'amp-social-share[type="twitter"]',
+				'event_name'     => 'social share',
+				'event_label'    => 'twitter',
+				'event_category' => 'NTG social',
+			],
```

Comment:
cross-referencing [`<amp-social-share>`](https://amp.dev/documentation/components/amp-social-share/)'s `type`s and [NTG guidelines](https://newsinitiative.withgoogle.com/training/states/ntg/assets/ntg-playbook.pdf#page=245), we're missing `whatsapp` and `linkedin`. Is there a reason there are only `facebook` and `twitter` here?

Reply:
This was intended to be the framework rather than a full implementation of the spec, so I just did a few elements as examples but I can add those.

Reply:
https://github.com/Automattic/newspack-plugin/pull/531/commits/5835e54b97a7bf369ca28be303fee3094ea139cb 

Note: The Jetpack WhatsApp share button isn't AMP-compatible so it won't show up in AMP, but when it's compatible we will already be prepared. :)